metaflow/plugins/metadata_providers/local.py

```diff
@@ -202,6 +203,110 @@ def _optimistically_mutate():
             "Tagging failed due to too many conflicting updates from other processes"
         )
 
+    @classmethod
+    def filter_tasks_by_metadata(
+        cls,
+        flow_id: str,
+        run_id: str,
+        query_step: str,
+        field_name: str,
+        field_value: str,
+    ) -> List[str]:
+        """
+        Filter tasks by metadata field and value, returning task IDs that match criteria.
+
+        Parameters
+        ----------
+        flow_id : str
+            Identifier for the flow
+        run_id : str
+            Identifier for the run
+        query_step : str
+            Name of the step to query tasks from
+        field_name : str
+            Name of metadata field to query
+        field_value : str
+            Value to match in metadata field
+
+        Returns
+        -------
+        List[str]
+            List of task IDs that match the query criteria
+
+        Raises
+        ------
+        JSONDecodeError
+            If metadata file is corrupted or empty
+        FileNotFoundError
+            If metadata file is not found
+        """
+
+        def _get_latest_metadata_file(path: str, field_prefix: str) -> tuple:
+            """Find the most recent metadata file for the given field prefix."""
+            # The metadata is saved as files with the format: sysmeta_<field_name>_<timestamp>.json
+            # and the artifact files are saved as: <attempt>_artifact__<artifact_name>.json
+            # We loop over all the JSON files in the directory and find the latest one
+            # that matches the field prefix.
+            json_files = glob.glob(os.path.join(path, "*.json"))
```

Comment:
we should be able to do a more efficient globbing so we don't have to filter by field_prefix later on. SOmething like `f"{field_prefix}*.json"`.